modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition_manager.py

```diff
@@ -223,6 +225,42 @@ def map_axis_partitions(
             **kwargs,
         )
 
+    @classmethod
+    def _update_partition_dimension_caches(cls, partitions: np.ndarray):
+        """
+        Build and set the length and width caches of each _physical_ partition in the given list of partitions.
+
+        Parameters
+        ----------
+        partitions : np.ndarray
+            The partitions for which to update length caches.
+        """
+        promises = []
+        # If `ray_idxs[i] = j`, that means `promises[i]` represents a computation corresponding
+        # to a dimension of `partitions[j]`.
+        part_idxs = []
+        # If `is_lens[i] = True`, then `promises[i]` is a computation for a length; otherwise
+        # it is for a width
+        is_lens = []
+        for i, part in enumerate(partitions):
+            l_cache = part.try_build_length_cache()
+            if isinstance(l_cache, ObjectIDType):
+                promises.append(l_cache)
+                part_idxs.append(i)
+                is_lens.append(True)
+            w_cache = part.try_build_width_cache()
+            if isinstance(w_cache, ObjectIDType):
+                promises.append(w_cache)
+                part_idxs.append(i)
+                is_lens.append(False)
+        new_dims = ray.get(promises)
+        for i, new_cache in enumerate(new_dims):
+            part = partitions[part_idxs[i]]
+            if is_lens[i]:
+                part.try_set_length_cache(new_dims[i])
```

Comment:
should `new_dims[i]` be `new_cache`?

Reply:
+1

Reply:
good catch, I forgot that's why I used the `enumerate`